providers/openmetrics/reporter.go

```diff
@@ -61,17 +61,16 @@ func (r *reporter) PostMsgReceive(_ interface{}, _ error, _ time.Duration) {
 }
 
 type reportable struct {
-	registry openmetrics.Registerer
+	clientMetrics *ClientMetrics
+	serverMetrics *ServerMetrics
 }
 
 func (rep *reportable) ServerReporter(ctx context.Context, _ interface{}, typ interceptors.GRPCType, service string, method string) (interceptors.Reporter, context.Context) {
-	m := NewServerMetrics(rep.registry)
```

Comment:
This is super weird. So we're creating a new metrics holder here, we do not register it anywhere and just use it as is. This is just broken - there is no way these metrics are going to be scraped.